src/shared/unit-file.c

```diff
@@ -31,6 +31,41 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_PATH);
 }
 
+int unit_dependency_name_compatible(const char *symlink, const char *target, bool instance_propagation) {
```

Comment:
a dependency name sounds like it would be the name of a dependency, i.e. one of Wants, Requires, Requisite, After, …

Reply:
Maybe call this "unit_symlink_name_compatible()"? It's exactly what we use it for, right?